### PR TITLE
Fix:  xpath=//input[@data-testid="resource-manage-project-modal-name"] failure

### PR DESCRIPTION
**Pull request by ConorOM1**
Failures discovered in ODH Smoke & Sanity

Refactored to identify the element by data-testid only.

Tests fixed:
ODS-1784
ODS-1786
ODS-1775

### CODE REVIEW

**Review by jiridanek — DISMISSED** [review]

File: ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot [hunk 1]
```diff
@@ -15,9 +15,9 @@ Test Tags          Dashboard
 
 
 *** Variables ***
-${PRJ_TITLE}=   ODS-CI Common Prj
+${PRJ_TITLE}=   ods-ci-common-project
```
Comment (jstourac) [inline ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot:18]: Is this change intentional as part of this fix? It now matches what the `PRJ_RESOURCE_NAME` contains.

Comment (jiridanek) [reply ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot:18]: Good point, looks I'm not suspecting enough

Comment (ConorOM1) [reply ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot:18]: yes intentional. was causing the following setup failure `ODS-CI Common Prj != ods-ci-common-project` as the project title used is now ods-ci-common-project

Comment (jstourac) [reply ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot:18]: I don't get it :slightly_smiling_face: This variable is to define the project title, no? So, what is causing it to be ods-ci-common-project? Isn't there some conflict with what is defined by the other variable I mentioned?

Comment (ConorOM1) [reply ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot:18]: hmmm yes good point. it looks like somewhere along the line ${PRJ_TITLE} is being changed ? Or not being checked correctly ?

File: ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource [hunk 2]
```diff
@@ -8,7 +8,7 @@ Resource       ./Workbenches.resource
 ${DS_PROJECT_TITLE}=    Data Science Projects
 ${TITLE_INPUT_XP}=    xpath=//input[@id="manage-project-modal-name"]
 ${DESCR_INPUT_XP}=    xpath=//textarea[@id="manage-project-modal-description"]
-${RESOURCE_INPUT_XP}=    xpath=//input[@data-testid="resource-manage-project-modal-name"]
+${RESOURCE_INPUT_XP}=    css:[data-testid="manage-project-modal-name"]
```
Comment (jstourac) [inline ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource:11]:
```suggestion
${RESOURCE_INPUT_XP}=    css:[data-testid="resource-manage-project-modal-name"]
```

File: ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot (hunk 1 above)
Comment (jstourac) [reply ods_ci/tests/Tests/0400__ods_dashboard/0410__ods_dashboard_projects/0410__ods_dashboard_projects.robot:18]: I commented a suggestion above what I think is the rootcause.

File: ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource (hunk 2 above)
Comment (jstourac) [reply ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource:11]: Also, are you sure that this change works fine? Changing from xpath to css (AFAIK equivalent for the `document.querySelector()` in the browser console) - will this work with all the current places this variable is being used now? :thinking:

**Review by jstourac — CHANGES_REQUESTED** [review]
Commented above; I'm not sure about the current state of these changes.

File: ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource (hunk 2 above)
Comment (ConorOM1) [reply ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource:11]: this is actually the root cause of previous comment. was pointing to the wrong selector, need to clean up

re css: this is the best way going forward according to @manosnoam , by looking for the specific data-testid only as it should be unique

Comment (jstourac) [reply ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource:11]: I understand the motivation; what I'm not sure is the `css:` prefix there - will it work just fine with the current places this is being used? Shouldn't we have `//` prefix there?

Disclaimer: I didn't test it and I know basically nothing about this and html and stuff around it.

Comment (jiridanek) [reply ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource:11]: > I understand the motivation; what I'm not sure is the css: prefix there - will it work just fine with the current places this is being used? Shouldn't we have //* prefix there?

This is just SeleniumLibrary syntax for XPaths and css selectors; not really web-stuff related https://robotframework.org/SeleniumLibrary/SeleniumLibrary.html#:~:text=can%20be%20implicit.-,Default%20locator%20strategy,-By%20default%2C%20locators

Comment (ConorOM1) [reply ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Projects.resource:11]: thanks for your help @jstourac & @jiridanek

**Review by jstourac — APPROVED** [review]

**Review by manosnoam — APPROVED** [review]

**Comment by manosnoam** [comment]
@ConorOM1 Thanks for the fix. 
BTW, Do you know how will it work without the square brackets around the CSS value ?

**Comment by ConorOM1** [comment]
> @ConorOM1 Thanks for the fix! BTW, Do you know how will it work without the square brackets around the CSS value ?

@manosnoam TY ! without brackets throws an invalid selector exception

**Review by jiridanek — APPROVED** [review]
